lib/mutant/matcher/method.rb

```diff
@@ -1,133 +1,154 @@
 module Mutant
   class Matcher
-    # Matcher for subjects that are a specific method
+    # Abstract base class for method matchers
     class Method < self
-      include Adamantium::Flat, Concord::Public.new(:env, :scope, :target_method)
-      include AST::NodePredicates
+      include AbstractType,
+              Adamantium::Flat,
+              Concord::Public.new(:scope, :target_method, :evaluator)
 
       # Methods within rbx kernel directory are precompiled and their source
       # cannot be accessed via reading source location. Same for methods created by eval.
       BLACKLIST = %r{\A(kernel/|\(eval\)\z)}.freeze
 
-      # Enumerate matches
+      SOURCE_LOCATION_WARNING =
+        '%s does not have valid source location, unable to emit subject'.freeze
+
+      CLOSURE_WARNING =
+         '%s is dynamically defined in a closure, unable to emit subject'.freeze
```

Comment:
In summary it detected that case:

```
class Foo
end

klass = some_cool_dynamic_computation_resulting_in_foo
klass.class_eval do
   def bar
   end
end
```

If this `Foo#bar` candidate for subject emitting on the expression `Foo#bar` mutant cannot do this, as per current runtime mutation insertion: Mutant would not know how to reproduce the constant scope `Foo#bar` is defined under. There is no monkeypatch syntax mutant could generate that by 100% would reproduce the same constant scope.